experiments/run_experiment.py

```diff
@@ -214,59 +232,153 @@ def _get_mrmc(
         rescale_direction=mrmc_method.get_constant_step_size_rescaler(
             step_size
         ),
-        random_seed=random_seed,
-        confidence_threshold=confidence_threshold,
+        confidence_threshold=confidence_cutoff,
+        model=model,
+        random_seed=cluster_seed,
+    )
+    iterator = recourse_iterator.RecourseIterator(
+        adapter=adapter,
+        recourse_method=mrmc,
+        certainty_cutoff=confidence_cutoff,
         model=model,
     )
 
+    # get the POI
+    poi = utils.random_poi(
+        dataset,
+        label_column=dataset_info.label_column,
+        label_value=adapter.negative_label,
+        model=model,
+        random_seed=poi_seed,
+    )
+
+    # generate the paths
+    paths = iterator.iterate_k_recourse_paths(
+        poi=poi, max_iterations=max_iterations
+    )
+
+    # retrieve the clusters
+    cluster_df = pd.DataFrame(
+        data=mrmc.clusters.cluster_centers,
+        columns=adapter.embedded_column_names(),
+    )
+
+    return paths, cluster_df
+
 
-def _get_recourse_iterator(
-    adapter: recourse_adapter.RecourseAdapter,
-    mrmc: mrmc_method.MRMC,
-    confidence_cutoff: float,
-    model: model_interface.Model,
-) -> recourse_iterator.RecourseIterator:
-    """Gets the recourse iterator. Useful for testing."""
-    return recourse_iterator.RecourseIterator(
+def run_dice(
+    run_seed: int,
+    confidence_cutoff: Optional[float],
+    noise_ratio: Optional[float],
+    rescale_ratio: Optional[float],
+    num_paths: int,
+    max_iterations: int,
+    dataset_name: str,
+    model_type: str,
+    **_unused_kwargs: Any,
+) -> Sequence[pd.DataFrame]:
+    """Runs DICE using the given configurations.
+
+    Args:
+        run_seed: The seed used in the run. All random numbers are derived from
+            this seed.
+        confidence_cutoff: The target model confidence.
```

Comment:
Confidence cutoff could be considered a hyperparameter for DiCE.

Reply:
This is true -- so could number of paths. I'll follow up with you about this.